server/manifest.go

```diff
@@ -57,6 +60,18 @@ func (m *Manifest) RemoveLayers() error {
 	return nil
 }
 
+func (m *Manifest) Validate() error {
+	if m.SchemaVersion != ManifestSchemaVersion {
```

Comment:
This is looking great!

We _may_ remove these two fields later: `.Config` is an unnecessarily-nested json blob, and I don't think `.SchemaVersion` is (or should be) used.

Do you know where we might depend on them later in the code (and could we remove the need for them?) I believe the `.Config` field is required, but maybe a missing `.SchemaVersion` could be silently ignored?

Reply:
You are correct that SchemaVersion is not used anywhere currently - it is a write only field. The current version is v2 and this dates back to the first commit - was some there some history that required a version bump from v1? If we are comfortable that we would always just version breaking changes by introducing new field names then we might as well remove it now and stop including it in newly published models. There would be no backwards compatibility issues.

From Config the only field that we use is Digest, which is used to find the config file. Of the fields in that file (which are things like arch, parameters, quantization, etc.), we do use them but mostly in response to commands like ollama show and I assume on the website. They are not actually needed for running the models as the GGUF files include essentially the same metadata. That being said, I don't think there is a good reason to try to de-duplicate these fields - it may limit our flexibility to use other model formats in the future.

Reply:
Thanks for hunting that down. That sounds good, it makes sense to remove the check for `SchemaVersion`

Leaving the check for config is okay for now, however I do wonder if we could instead effectively treat it as an optional field and fail gracefully when we try to access it via `.Config.Digest` elsewhere in the code (vs making it a hard required field here on pull, etc).

That would give us a chance to deprecate it later if we wanted to without breaking today's version of Ollama (the general idea going forward would be to put metadata in the manifest object itself vs in a `.Config` "layer" since it's redundant to have to go fetch another layer for a basic json object). Let me know if that makes sense

Reply:
Here's another try that addresses the following issues:
 * Cleans up the error messages
 * Removes SchemaVersion (both the check and writing it at all in new manifests)
 * Allows .Config to be missing, with just the loss of some information printed in ollama show

The last one has a few caveats if we actually do remove the config from a layer and put it into the manifest file itself. However, none of them are worse than the alternatives - in those cases, we wouldn't be able to make this change at all.
 * Clients that predate this patch (i.e. all of them today) will crash as it will trigger the bug as originally reported here. Obviously we can put the config in both places for a while to stage the rollout and eventually old versions will be thoroughly obsolete anyways.
 * For clients that are new enough to not crash but old enough to not understand the new manifest format, there can be some information loss if they download and then re-upload a model.

Lastly, code-wise this is a little ugly with a bunch of checks for missing config scattered around. I changed the struct so that it is not a pointer so Ollama won't crash if a check is missed, either now or in the future.

Let me know if you have any other suggestions.

Reply:
This seems right! Thanks!